File: 0189-rotate-array/0189-rotate-array.java
class Solution {
    public void rotate(int[] nums, int k) {
        Deque<Integer> dq = new ArrayDeque();
        for(int i:nums) {
            dq.offer(i);
        }
        while(k-- > 0) {
            dq.offerFirst(dq.pollLast());
        }
        for(int i=0; i<nums.length; i++) {
            nums[i] = dq.poll();
        }
    }
}

File: 0189-rotate-array/0189-rotate-array.py
class Solution:
    def rotate(self, nums: List[int], k: int) -> None:
        k %= len(nums)
        if len(nums) == k: return
        nums[:len(nums)-k] = nums[len(nums)-k-1::-1]
        nums[len(nums)-k:len(nums)] = nums[len(nums):len(nums)-k-1:-1]
        nums[::] = nums[::-1]